src/renderers/webgl/WebGLLights.js

```diff
@@ -306,15 +312,27 @@ function WebGLLights() {
 					shadowUniforms.shadowRadius = shadow.radius;
 					shadowUniforms.shadowMapSize = shadow.mapSize;
 
-					state.spotShadow[ spotLength ] = shadowUniforms;
-					state.spotShadowMap[ spotLength ] = shadowMap;
-					state.spotShadowMatrix[ spotLength ] = light.shadow.matrix;
+					if ( map ) {
+
+						state.spotShadow[ spotLength ] = state.spotShadow[ spotMapLength ];
```

Comment:
Can you please add a comment that describes the assignments of this block. TBH, I have a hard time to understand what you are doing here...

Reply:
I added a comment : 
`// keep lights with a map at the front of the state arrays`

Reply:
I finally changed that to support maps with no shadows. the sorting is doing once and for all (with/without maps and with/without shadows) by the lights.sort function call